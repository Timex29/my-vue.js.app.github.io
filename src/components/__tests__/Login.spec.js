import { describe, test, expect } from 'vitest'
import { mount } from '@vue/test-utils'


import LoginComponent from '../UserLogin.vue'

describe('Login page attributes', () => {
  test('renders properly @timex', async () => {
    const wrapper = mount(LoginComponent)

    expect(LoginComponent).toMatchSnapshot()

    //const username = 'timex'
    //const password = ''

    //await wrapper.find('input[type=text]').setValue(username)
    //await wrapper.find('input[type=password]').setValue(password)

    //await wrapper.find('form').trigger('submit.prevent')
    //await wrapper.find('.link a')
    

   

    //expect(wrapper.emitted('submit')[0][0]).toStrictEqual({
      //username,
      //password
    //})

    expect(await wrapper.html()).toContain('Sign in')
    expect(wrapper.find('.box').exists()).toBe(true)
    expect(wrapper.find('.input').exists()).toBe(true)
    expect(wrapper.find('.inputBox').exists()).toBe(true)

   


    await wrapper.find('[ data-type = "submit"]').trigger('click')
   
  })
  test('throws an error for non existing users', async () => {
    expect(async () => await LoginComponent('fakeuser')).rejects.toThrowError()
  
  })
  
})

