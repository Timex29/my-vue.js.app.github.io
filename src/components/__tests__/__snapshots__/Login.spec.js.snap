// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login page > renders properly @timex 1`] = `
{
  "__file": "F:/programs_installed/files/github-clone/with-vuetify/my-vue.js.app.github.io/src/components/UserLogin.vue",
  "data": [Function],
  "methods": {
    "onSubmit": [Function],
    "required": [Function],
  },
  "render": [Function],
}
`;

exports[`Login page attributes > renders properly @timex 1`] = `
{
  "__file": "F:/programs_installed/files/github-clone/with-vuetify/my-vue.js.app.github.io/src/components/UserLogin.vue",
  "data": [Function],
  "methods": {
    "onSubmit": [Function],
    "required": [Function],
  },
  "render": [Function],
}
`;

exports[`Login pageattri > renders properly @timex 1`] = `
{
  "__file": "F:/programs_installed/files/github-clone/with-vuetify/my-vue.js.app.github.io/src/components/UserLogin.vue",
  "data": [Function],
  "methods": {
    "onSubmit": [Function],
    "required": [Function],
  },
  "render": [Function],
}
`;

exports[`Login pageattru > renders properly @timex 1`] = `
{
  "__file": "F:/programs_installed/files/github-clone/with-vuetify/my-vue.js.app.github.io/src/components/UserLogin.vue",
  "data": [Function],
  "methods": {
    "onSubmit": [Function],
    "required": [Function],
  },
  "render": [Function],
}
`;
